File: InsertAtEnd.java
// public class InsetAtBeginning {
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }

}

class linkedList {
    Node head;
    Node tail;

    linkedList() {
        head = null;
        tail = null;
    }

    public void insertAtBeginning(int data) {
        Node x = new Node(data);
        x.next = this.head;
        this.head = x;
        if (tail == null) {
            tail = x;
        }

    }

    public void insertAtEnd(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;

        } else {
            tail.next = newNode;
        }
        tail = newNode;
        // Node current = head;
        // while (current.next != null) {
        // current = current.next;
        // }
        // current.next = newNode;
    }

    public void display() {
        Node temp = head;
        while (temp != null) {
            System.out.print(temp.data + " ");
            temp = temp.next;
        }
    }
}

public class InsertAtEnd {

    public static void main(String[] args) {
        linkedList ll = new linkedList();
        ll.insertAtBeginning(5);
        ll.insertAtBeginning(12);
        ll.insertAtBeginning(53);
        ll.insertAtBeginning(50);
        ll.insertAtBeginning(55);
        ll.insertAtBeginning(51992);
        ll.insertAtEnd(23);
        ll.insertAtEnd(2400);
        ll.insertAtEnd(250);

        ll.display();
    }
}

File: InsertAtEnd.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None
        self.tail = None

    def insert_at_beginning(self, data):
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node
        if self.tail is None:
            self.tail = new_node

    def insert_at_end(self, data):
        new_node = Node(data)
        if self.head is None:
            self.head = new_node
            self.tail = new_node
        else:
            self.tail.next = new_node
            self.tail = new_node

    def display(self):
        temp = self.head
        while temp is not None:
            print(temp.data, end=" ")
            temp = temp.next
        print() 


ll = LinkedList()
ll.insert_at_beginning(5)
ll.insert_at_beginning(12)
ll.insert_at_beginning(53)
ll.insert_at_beginning(50)
ll.insert_at_beginning(55)
ll.insert_at_beginning(51992)
ll.insert_at_end(23)
ll.insert_at_end(2400)
ll.insert_at_end(250)
ll.insert_at_end(252)
ll.display() 